### PR TITLE
Orders command

### LINKED ISSUE #6: List orders command
List orders command

### PR DESCRIPTION
**Pull request by anxolin**
Closes #6 

Basic command for getting orders.

When an order hasn't been traded, looks like this:
image

If there's trades for the order, it looks like this:
image

Deleted or canceled orders looks like this:
image

Out of this PR: https://github.com/gnosis/dex-cli/issues/31

### CODE REVIEW

File: src/commands/orders.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+          calculate_price(
+            numerator=soldVolume,
+            denominator=boughtVolume,
+            decimals_numerator=sellTokenDecimals,
+            decimals_denominator=buyTokenDecimals
+          ),
+          currency=sellTokenLabel
+        ) +
+        '\n'
+      )
```
Comment (alfetopito) [inline src/commands/orders.py]: Nit: I would put the prices into their own variables and simplify the string building to something like:

```python
price = format_price(
          calculate_price(
            numerator=boughtVolume,
            denominator=soldVolume,
            decimals_numerator=buyTokenDecimals,
            decimals_denominator=sellTokenDecimals
          ), 
          currency=buyTokenLabel
        )
priceStr1 = f'{click.style(f'  Avg. Traded Price {sellTokenLabel}/{buyTokenLabel}', fg=labelColor)}: {price1}'

...

tradePriceText = f'{priceStr1}\n{priceStr2}\n'
```

File: src/commands/orders.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+      click.style('  Sold volume', fg=labelColor) + ': ' + 
+      format_amount_in_weis(soldVolume, sellTokenDecimals) + ' ' + sellTokenLabel + '\n' + 
+
+      (
+        click.style('  Bought volume', fg=labelColor) + ': ' + 
+        format_amount_in_weis(boughtVolume, buyTokenDecimals) + ' ' + buyTokenLabel + '\n'
+        if soldVolume else ''
+      ) + # TODO: Add percentage https://github.com/gnosis/dex-cli/issues/31
+      '\n' +
```
Comment (alfetopito) [inline src/commands/orders.py]: Would it be too much to also add the order size? Meaning, max buy/sell amounts, not just the matched amounts.

File: src/utils.py [hunk 3]
```diff
@@ -29,28 +37,71 @@ def format_token_short(token):
 
 
 def format_integer(number):
-  return str(number) # TODO: Format better the numbers
+  return '{:,d}'.format(number)
```
Comment (alfetopito) [inline src/utils.py]: Nit: you can also use f strings for that https://stackoverflow.com/a/52650045/1272513

File: src/utils.py [hunk 4]
```diff
@@ -29,28 +37,71 @@ def format_token_short(token):
...
-  return str(amount / (10 ** decimals)) # TODO: Format better the weis
+def format_price(amount, decimals=10, rounding=ROUND_DOWN, currency=''):
+  price = format_amount(amount, decimals=decimals, rounding=rounding)
+  return price + ' ' + currency if currency else price
+
+def format_amount(amount, decimals=18, rounding=ROUND_DOWN):
+  quantize_value = Decimal(10) ** -Decimal(decimals)
+  rounded_value = Decimal(amount).quantize(quantize_value, rounding=rounding)
+
+  return f'{{:,.{decimals}f}}'.format(rounded_value).rstrip('0').rstrip('.')
```
Comment (alfetopito) [inline src/utils.py]: Yikes, f string and format?

File: src/utils.py [hunk 5]
```diff
@@ -29,28 +37,71 @@ def format_token_short(token):
...
-def format_date_time(date):
-  return '' if date is None else date.strftime("%d/%m/%y %H:%M:%S")
+def format_date_time(date, tooBigLabel='Never'):  
+    return tooBigLabel if date == datetime.max else date.strftime("%d/%m/%y %H:%M:%S")
+
+def format_batch_id_with_date(batchId, tooBigLabel='Never expires'):
+  if batchId:
+    return tooBigLabel if batchId >= MAX_BATCH_ID else (
+      format_integer(batchId) + 
+      f" ({ format_date_time(to_date_from_batch_id(batchId))})"
```
Comment (alfetopito) [inline src/utils.py]:
```suggestion
      f"{format_integer(batchId)} ({ format_date_time(to_date_from_batch_id(batchId))})"
```

File: src/utils.py [hunk 6]
```diff
@@ -29,28 +37,71 @@ def format_token_short(token):
...
+  if denominator_dec.is_zero():
+    return Decimal(1) if numerator_dec.is_zero() else Infinity
+  else:
+    precision_factor = Decimal(10) ** Decimal(abs(decimals_numerator - decimals_denominator))
+    if decimals_numerator > decimals_denominator:
+      return numerator_dec / denominator_dec / precision_factor
+    else:
+      return numerator_dec / (denominator_dec / precision_factor)
+
+  if decimals_numerator > decimals_denominator:
```
Comment (alfetopito) [inline src/utils.py]: Is this reachable?

File: src/commands/orders.py (hunk 2 above)
Comment (anxolin) [reply src/commands/orders.py]: This was added in a different PR. I added sell amount. Take a look if you miss the max receiving amount. Probably we should add it as you say

File: src/commands/orders.py (hunk 1 above)
Comment (anxolin) [reply src/commands/orders.py]: https://github.com/gnosis/dex-cli/issues/39

File: src/commands/orders.py (hunk 2 above)
Comment (anxolin) [reply src/commands/orders.py]: https://github.com/gnosis/dex-cli/issues/39

File: src/utils.py (hunk 3 above)
Comment (anxolin) [reply src/utils.py]: Great tip!!!

https://github.com/gnosis/dex-cli/issues/39

File: src/utils.py (hunk 4 above)
Comment (anxolin) [reply src/utils.py]: Didn't know about the formatting. I'll address in https://github.com/gnosis/dex-cli/issues/39

File: src/utils.py (hunk 6 above)
Comment (anxolin) [reply src/utils.py]: O myyyyy! I forgor to delete that last part! I'll address it here https://github.com/gnosis/dex-cli/issues/39

super good catch!

File: src/utils.py (hunk 5 above)
Comment (anxolin) [reply src/utils.py]: Also here: https://github.com/gnosis/dex-cli/issues/39

**Pull request merged**